pkg/controller/storagecluster/cephcluster.go

```diff
@@ -519,11 +520,16 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 // The checkTuneStorageDevices function checks whether devices from the given
 // storage class are a known type that should expclitly be tuned for fast or
 // slow access.
-func (r *ReconcileStorageCluster) checkTuneStorageDevices(storageClassName string) (diskSpeed, error) {
+func (r *ReconcileStorageCluster) checkTuneStorageDevices(ds ocsv1.StorageDeviceSet) (diskSpeed, error) {
+	if DeviceTypeHDD == strings.ToLower(ds.DeviceType) {
+		return diskSpeedSlow, nil
+	} else if DeviceTypeSSD == strings.ToLower(ds.DeviceType) || DeviceTypeNVMe == strings.ToLower(ds.DeviceType) {
+		return diskSpeedFast, nil
+	}
 	storageClass := &storagev1.StorageClass{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassName}, storageClass)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: *ds.DataPVCTemplate.Spec.StorageClassName}, storageClass)
```

Comment:
please assign `storageClassName := *ds.DataPVCTemplate.Spec.StorageClassname`. This reduces the diff, and the var is used more than once...